messaging/src/main/java/org/axonframework/eventhandling/tokenstore/TokenStore.java

```diff
@@ -207,6 +211,25 @@ default boolean requiresExplicitSegmentInitialization() {
      */
     int[] fetchSegments(String processorName);
 
+    /**
+     * Returns a List of known available {@code segments} for a given {@code processorName}. A segment is considered available if it is not claimed by any
+     * other event processor.
+     * <p>
+     * The segments returned are segments for which a token has been stored previously and have not been claimed by another processor. When the {@link
+     * TokenStore} is empty, an empty list is returned.
+     *
+     * By default, if this method is not implemented, we will return all segments instead, whether they are available or not.
+     *
+     * @param processorName the processor's name for which to fetch the segments
+     * @return a List of available segment identifiers for the specified {@code processorName}
+     */
+    default List<Segment> fetchAvailableSegments(String processorName) {
+        int[] allSegments = fetchSegments(processorName);
+        return Arrays.stream(allSegments).boxed()
+                .map(segment -> Segment.computeSegment(segment, allSegments))
```

Comment:
Nit: the indentation is slightly off here. I'd expect the `map` operation to be on the same line as the `stream` operation.